File: app/src/main/java/com/harrysoft/burstcoinexplorer/burst/entity/SearchResult.java
package com.harrysoft.burstcoinexplorer.burst.entity;

public class SearchResult {
    public final String request;
    public final SearchRequestType requestType;

    private boolean navigated = false;

    public SearchResult(String request, SearchRequestType requestType) {
        this.request = request;
        this.requestType = requestType;
    }

    public boolean isNavigated() {
        return navigated;
    }

    public void onNavigated() {
        navigated = true;
    }
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/entity/SearchResult.kt
package com.harry1453.burst.explorer.entity

data class SearchResult (
        @JvmField val request: String,
        @JvmField val requestType: SearchRequestType
) {
    private var navigated: Boolean = false

    fun isNavigated(): Boolean {
        return navigated
    }

    fun onNavigated() {
        navigated = true
    }
}
